### ISSUE TITLE
fatal errors in git flow init

### ISSUE THREAD

**Issue by alekam** [nvie/gitflow#18]
I follow instructions in README file, use 'git flow init' instead of 'git init' and get fatal errors (RHEL5):

$ git flow init

fatal: Not a git repository
fatal: Not a git repository: ''
fatal: Not a git repository: ''
fatal: Not a git repository: ''
fatal: Not a git repository: ''
Not a git repository

$ git flow help init

fatal: Not a git repository
fatal: Not a git repository: ''
fatal: Not a git repository: ''
fatal: Not a git repository: ''
usage: git flow <subcommand>

Available subcommands are:
   init      Initialize a new git repo with support for the branching model.
   feature   Manage your feature branches.
   release   Manage your release branches.
   hotfix    Manage your hotfix branches.
   support   Manage your support branches.
   version   Shows version information.

Try 'git flow <subcommand> help' for details.

**Comment by bruntonspall** [comment]
Hello,

I've fixed this in a version in my tree.  Please try cloning bruntonspall/gitflow and using the feature/init branch and let me know if it fixes it.

**Comment by nvie** [comment]
Thanks for the bug report and thanks, bruntonspell, for the patch.

To be honest, git-flow init hasn't gotten the attention it should have. I have some plans for the git-flow init subcommand to have it query interactively for some of the configurable properties of git-flow repos.

Furthermore, I would like to design the behaviour of it such that it could be run inside:
- an empty directory (i.e. no git repo yet);
- a regular, init'ed git repo;
- an existing project that not necessarily has a branch structure that resembles gitflow's required structure (init can offer to fix this)

Nevertheless, I have borrowed the concept of bruntonspall's patch, but I made some changes to it. For example, I moved the initialize() function to gitflow-common and made the subcommands explicitly invoke it if they need it (they all do except for init). This way, you don't have to make an explicit exception within the main git-flow script for the init subcommand. This is a somewhat cleaner implementation in my opinion.
